bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -305,31 +331,48 @@ public BigtableSession(BigtableOptions opts) throws IOException {
   }
 
   // <editor-fold desc="Interceptors">
+
+  /**
+   * @deprecated Channel creation is now considered an internal implementation detail channel
+   *     creation methods will be removed from the public surface in the future
+   */
+  @Deprecated
   protected List<ClientInterceptor> setupInterceptors() throws IOException {
-    List<ClientInterceptor> clientInterceptorsList = new ArrayList<>();
-    clientInterceptorsList.add(
-        new GoogleCloudResourcePrefixInterceptor(options.getInstanceName().toString()));
-    clientInterceptorsList.add(createGaxHeaderInterceptor());
+    List<ClientInterceptor> clientInterceptorsList = new ArrayList<>(createInterceptors(options));
+    clientInterceptorsList.add(setupWatchdog());
+
+    return clientInterceptorsList;
+  }
+
+  @InternalApi
+  static List<ClientInterceptor> createInterceptors(BigtableOptions options) throws IOException {
+    ImmutableList.Builder<ClientInterceptor> interceptors = ImmutableList.builder();
+
+    // TODO: instanceName should never be null
```

Comment:
should we throw here instead?

Reply:
In the future yes. But for now no. This PR should retain existing behavior